src-tauri/src/commands.rs

```diff
@@ -380,3 +402,419 @@ pub async fn close_region_selector(app: AppHandle) -> Result<(), String> {
+    
+    // Validate the path is within the output folder
+    if !file_path.starts_with(&folder) {
+        return Err("Invalid file path".to_string());
+    }
+    
+    if file_path.exists() {
+        std::fs::remove_file(&file_path)
+            .map_err(|e| format!("Failed to delete file: {}", e))?;
+        
+        // Also delete thumbnail if it exists
+        let thumbnail_path = file_path.with_extension("").to_string_lossy().to_string() + "_thumb.jpg";
+        if std::path::Path::new(&thumbnail_path).exists() {
+            let _ = std::fs::remove_file(&thumbnail_path);
+        }
+        
+        // Renumber remaining recordings
+        renumber_recordings(&folder)?;
+        
+        // Update clip count
+        {
+            let mut count = state.clip_count.lock().unwrap();
+            *count = count.saturating_sub(1);
+        }
+        
+        println!("🗑️ Deleted recording: {}", filename);
+    }
+    
+    Ok(())
+}
+
+/// Renumber recordings sequentially after deletion
+fn renumber_recordings(folder: &std::path::Path) -> Result<(), String> {
+    let mut recordings = Vec::new();
+    
+    if let Ok(entries) = std::fs::read_dir(folder) {
+        for entry in entries.flatten() {
+            if let Some(file_name) = entry.file_name().to_str() {
+                if file_name.ends_with(".mp4") {
+                    // Check if the filename (without extension) is purely numeric
+                    let name_without_ext = file_name.trim_end_matches(".mp4");
+                    if name_without_ext.chars().all(|c| c.is_ascii_digit()) {
+                        if let Ok(metadata) = entry.metadata() {
+                            recordings.push((entry.path(), metadata.created().unwrap_or_else(|_| std::time::SystemTime::UNIX_EPOCH)));
+                        }
+                    }
+                }
+            }
+        }
+    }
+    
+    // Sort by creation time (oldest first)
+    recordings.sort_by(|a, b| a.1.cmp(&b.1));
+    
+    // Rename files sequentially
+    for (i, (old_path, _)) in recordings.iter().enumerate() {
+        let new_name = format!("{}.mp4", i + 1);
+        let new_path = folder.join(&new_name);
+        
+        if old_path != &new_path {
+            std::fs::rename(old_path, &new_path)
+                .map_err(|e| format!("Failed to rename file: {}", e))?;
+            
+            // Also rename thumbnail if it exists
+            let old_thumb = old_path.with_extension("").to_string_lossy().to_string() + "_thumb.jpg";
+            let new_thumb = new_path.with_extension("").to_string_lossy().to_string() + "_thumb.jpg";
+            if std::path::Path::new(&old_thumb).exists() {
+                let _ = std::fs::rename(&old_thumb, &new_thumb);
+            }
+        }
+    }
+    
+    Ok(())
+}
+
+/// Generate thumbnail for a recording using FFmpeg command
+#[tauri::command]
+pub fn generate_thumbnail(video_path: String) -> Result<String, String> {
```

Comment:
Damn, I got rid of ffmpeg and you brought it back!  ;-)

https://github.com/mieweb/pulse-desktop/commit/ecf819445ef873ab527999965ae0deef9ce8b646